webapp/models.py

```diff
@@ -90,43 +90,63 @@ def _process_html(html):
 
 def _replace_notifications(soup):
     """
-    Given some BeutifulSoup of a document,
+    Given some BeautifulSoup of a document,
     replace blockquotes with the appropriate notification markup
     """
 
     with open("templates/_notification.html") as notification_file:
         notification_template = Template(notification_file.read())
 
-    for note in soup.findAll(text=re.compile("ⓘ ")):
-        paragraph = note.parent
-        blockquote = paragraph.parent
-
-        if paragraph.name == "p" and blockquote.name == "blockquote":
-            text_contents = paragraph.encode_contents().decode("utf-8")
-            new_contents = re.sub("^ⓘ ", "", text_contents)
+    for note_string in soup.findAll(text=re.compile("ⓘ ")):
+        first_paragraph = note_string.parent
+        blockquote = first_paragraph.parent
+        last_paragraph = blockquote.findChildren(recursive=False)[-1]
+
+        if first_paragraph.name == "p" and blockquote.name == "blockquote":
+            # Remove extra padding/margin
+            first_paragraph.attrs["class"] = "u-no-padding--top"
+            if last_paragraph.name == "p":
+                if "class" in last_paragraph.attrs:
```

Comment:
I feel like this conditional logic can be abstracted out into a "addClass" function which would add the necessary spacing. It can then be reused throughout this file....and I'm assuming elsewhere in this project.

Reply:
Yes it could. It would be nice if BS4 provided that. But it seems like over-engineering to me here, as, at the moment, we do it in exactly one place.

I hope if someone were to come in and copy that functionality for use again they might then consider abstracting it.

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it, as @barrymcgee reminded me of yesterday.